### PR TITLE
feat(kafka-streams): adds support for flatMap.

### PR DESCRIPTION
**Pull request by jcchavezs**
This PR adds support for flatMap in kafka-streams instrumentation. Related to https://github.com/openzipkin/brave/issues/952

Ping @jeqo @jorgheymans

### CODE REVIEW

File: instrumentation/kafka-streams/src/main/java/brave/kafka/streams/KafkaStreamsTracing.java [hunk 1]
```diff
@@ -263,6 +263,28 @@ public KafkaStreams kafkaStreams(Topology topology, Properties streamsConfig) {
...
 
+  /**
+   * Create a flatMap transformer, similar to {@link KStream#flatMap(KeyValueMapper)}, where its mapper
+   * action will be recorded in a new span with the indicated name.
+   *
+   * <p>Simple example using Kafka Streams DSL:
+   * <pre>{@code
+   * StreamsBuilder builder = new StreamsBuilder();
+   * builder.stream(inputTopic)
+   *        .transform(kafkaStreamsTracing.flatMap("myflatMap", (k, v) -> ...)
```
Comment (jeqo) [inline instrumentation/kafka-streams/src/main/java/brave/kafka/streams/KafkaStreamsTracing.java]:
```suggestion
   *        .flatTransform(kafkaStreamsTracing.flatMap("myflatMap", (k, v) -> ...)
```

File: instrumentation/kafka-streams/README.md [hunk 2]
```diff
@@ -68,9 +80,9 @@ Typically, there are at least two spans involved in traces produces by a Kafka S
 * One created by the Consumers that starts a Stream or Table, by `builder.stream(topic)`.
 * One created by the Producer that sends a records to a Stream, by `builder.to(topic)`
 
-By receiving records in a Kafka Streams application with Tracing enabled, the span created once
-a record is received will inject the span context on the headers of the Record, and it will get
-propagated downstream on the Stream topology. As span context is stored in the Record Headers,
+By receiving records in a Kafka Streams application with Tracing enabled, the span created, once
```
Comment (jorgheymans) [inline instrumentation/kafka-streams/README.md]: good to revise that sentence but it still does not make so much sense. Not blocking this PR though.

**Comment by jorgheymans** [comment]
LGTM :+1:

**Comment by jcchavezs** [comment]
@jorgheymans or @jeqo could you do the honors?

**Comment by jeqo** [comment]
Thanks jc! Will rephrase and improve docs in a new PR.

**Pull request merged**